packages/flutter/test/widgets/slotted_render_object_widget_test.dart

```diff
@@ -139,6 +139,31 @@ void main() {
     expect(_RenderTest().publicNameForSlot(slot), slot.toString());
   });
 
+  testWidgets('key reparenting', (WidgetTester tester) async {
+    const Widget widget1 = SizedBox(key: ValueKey<String>('smol'), height: 10, width: 10);
+    const Widget widget2 = SizedBox(key: ValueKey<String>('big'), height: 100, width: 100);
+
+    await tester.pumpWidget(buildWidget(topLeft: widget1, bottomRight: widget2));
+    final _RenderDiagonal renderObject = tester.renderObject(find.byType(_Diagonal));
+    expect(renderObject._topLeft!.size, const Size(10, 10));
+    expect(renderObject._bottomRight!.size, const Size(100, 100));
+
+    // Swapping.
+    await tester.pumpWidget(buildWidget(topLeft: widget2, bottomRight: widget1));
+    expect(renderObject._topLeft!.size, const Size(100, 100));
```

Comment:
These tests need to assert that the elements backing the widgets are still the same after the swap/move. That's the main indicator that something was actually moved and not just destroyed and recreated.